### PR TITLE
docs: add release notes for 0.33.0

### PR DESCRIPTION
**Pull request by billboggs**
Add release notes for 0.33.0.

### CODE REVIEW

File: docs/release-notes.rst [hunk 1]
```diff
@@ -6,10 +6,91 @@
...
+      possible.
+
+**Improvements**
+
+Kubernetes: Add Determined resource information such as ``workspace`` and ``task ID`` as pod labels.
+This improvement facilitates better resource tracking and management within Kubernetes environments.
+
+Configuration: Introduce a DCGM Helm chart and Prometheus configuration to the
+``tools/observability`` directory. Additionally, two new dashboards, "API Monitoring" and "Resource
+Utilization", have been added to improve observability and operational insight.
```
Comment (hkumar92) [inline docs/release-notes.rst:41]: Have been added to...where? Grafana right?

**Review by hkumar92 — CHANGES_REQUESTED** [review]

File: docs/release-notes.rst (hunk 1 above)
Comment (billboggs) [reply docs/release-notes.rst]: Yeah after looking at the code change, they've been added to Grafana. I updated the wording to explicitly call that out.

Comment (hkumar92) [reply docs/release-notes.rst]: Oh unless you mean added to the tools/observability directory

Comment (hkumar92) [reply docs/release-notes.rst]: If so, I'm good, apologies I didn't read it correctly

File: docs/release-notes.rst [hunk 2]
```diff
@@ -6,10 +6,91 @@
...
+**Breaking Changes**
+
+-  Helm: An entry for ``initialUserPassword`` is now required when running ``helm install``.
+   Existing deployments are unaffected. See :ref:`Helm Chart <helm-config-reference>`.
+
+-  Web UI: Enforce password requirements for all new non-remote users: see
+   :ref:`password-requirements`
+
+   -  This applies to users created using the ``Add User`` button in the Web UI for admins.
+
```
Comment (tara-det-ai) [inline docs/release-notes.rst:27]: WebUI

File: docs/release-notes.rst [hunk 3]
```diff
@@ -6,10 +6,91 @@
...
+   -  Admins may also change the passwords of other users with this same interface.
+
+   -  This does not affect logging in with an existing user who already has an empty or
+      non-compliant password, but we recommend setting good passwords for such users as soon as
+      possible.
+
```
Comment (tara-det-ai) [inline docs/release-notes.rst:33]: - **WebUI**: Enforce password requirements for all new non-remote users. See :ref:`password-requirements` for details.

   - Applies to users created using the **Add User** button in the WebUI for admins.
   - Admins can change the passwords of other users using the same interface.
   - Does not affect existing users with empty or non-compliant passwords, but setting strong passwords for these users is recommended.

Comment (billboggs) [reply docs/release-notes.rst:30]: Updated

**Pull request merged**